File: c/classwork/8-11-2024/perfect_num.c
// WAP to check perfect number using function
#include<stdio.h>
void perfect_num(int num){
int factorial = 1,sum=0;
    for(int i=num-1; i > 0; i--) {
        if(num%i == 0){
            sum+=i;
        }
    }
    num == sum ? printf("perfect number") : printf("not a perfect number");
     



}

void main(){
    int n;
    printf("enter a number:");
    scanf("%d",&n);
    perfect_num(n);
} 

File: python/classwork/perfect_num.py
n = int(input("Enter a num: "))
i=n-1
sum=0
while i>0:
    if n%i == 0:
        sum = sum+i
    i = i-1

if(n == sum):
    print("Perfect Number")
else:
    print("Not a perfect Number")
